src/bhv_basic_move.cpp

```diff
@@ -105,13 +107,21 @@ Bhv_BasicMove::execute( PlayerAgent * agent )
     agent->debugClient().setTarget( target_point );
     agent->debugClient().addCircle( target_point, dist_thr );
 
-    if ( ! Body_GoToPoint( target_point, dist_thr, dash_power
-                           ).execute( agent ) )
-    {
-        Body_TurnToBall().execute( agent );
+
+    if ( Body_GoToPoint( target_point, dist_thr, dash_power
+			 ).execute( agent ) ||
+	 Body_TurnToBall().execute( agent ) ) {
```

Comment:
This is technically correct given the short-circuit evaluation of || operator, however it's significantly more confusing than the original.